test/test.dart

```diff
@@ -465,13 +465,20 @@ Future<void> tryDeleteRealm(String path) async {
     return;
   }
 
+  final dummy = File("");
+  const duration = Duration(milliseconds: 100);
   for (var i = 0; i < 5; i++) {
     try {
       Realm.deleteRealm(path);
-      await File('$path.lock').delete();
+      
+      //delete lock file
+      await File('$path.lock').delete().onError((error, stackTrace) => dummy);
+
+      //delete compaction space file
+      await File('$path.tmp_compaction_space').delete().onError((error, stackTrace) => dummy);
```

Comment:
Why do we hide the errors when deleting these files?

Reply:
these files might not exists. So this is operatio is "try deleting the file and ingore if it exists" In one operation. Otherwise I will need check if the file exists first and then delete it, which is two operations to the file system which should be slower than just trying and ignoring the error.

Reply:
Why is this file not automatically deleted by Core? It sounds like a bug that we need to clean it up ourselves.

Reply:
Isn't it better to delete inside the implementation it Realm.compact? So that our users will not have to care about it.

Reply:
Core should not leave these around. If it does, it's a bug on their end.